tracing/src/lib.rs

```diff
@@ -287,6 +287,17 @@
 //! # }
 //!```
 //!
+//! Fields may have :'s in their names, but this may not be used with the local
+//! variable shorthand as : is not a valid rust pathname.
+//! ```
+//! # use tracing::{span, Level};
+//! # fn main() {
+//! // records an event with an : containing fieldname:
+//! //  - "guid:x-request-id", with the value abcdef
+//! span!(Level::TRACE, "api", "guid:x-request-id" = "abcdef");
+//! # }
```

Comment:
Rather than discussing colons in particular, I might reword this to just discuss fields that are not valid Rust identifiers, in general:
```suggestion
//! Fields with names that are not Rust identifiers, or with names that are Rust reserved words,
//! may be created using quoted string literals. However, this may not be used with the local
//! variable shorthand.
//! ```
//! # use tracing::{span, Level};
//! # fn main() {
//! // records an event with fields whose names are not Rust identifiers
//! //  - "guid:x-request-id", containing a `:`, with the value "abcdef"
//! //  - "type", which is a reserved word, with the value "request"
//! span!(Level::TRACE, "api", "guid:x-request-id" = "abcdef", "type" = "request");
//! # }
```

Reply:
done